unitTests/ui/SmartSnippetTest.ts

```diff
@@ -219,6 +219,30 @@ export function SmartSnippetTest() {
           expectChildren(test.cmp.element, [ClassNames.ANSWER_CONTAINER_CLASSNAME, UserFeedbackBannerClassNames.ROOT_CLASSNAME]);
         });
 
+        it('the thank you banner should not be active', () => {
+          expect(getFirstChild(UserFeedbackBannerClassNames.THANK_YOU_BANNER_ACTIVE_CLASSNAME)).toBeNull();
+        });
+
+        it('after clicking yes or no, the thank you banner should be active', () => {
+          getFirstChild(UserFeedbackBannerClassNames.NO_BUTTON_CLASSNAME).click();
+          expect(getFirstChild(UserFeedbackBannerClassNames.THANK_YOU_BANNER_ACTIVE_CLASSNAME)).not.toBeNull();
+        });
+
+        it('after clicking yes or no and receiving a new answer, the thank you banner should not be active', async done => {
+          getFirstChild(UserFeedbackBannerClassNames.NO_BUTTON_CLASSNAME).click();
+          await triggerQuestionAnswerQuery(false);
+          expect(getFirstChild(UserFeedbackBannerClassNames.THANK_YOU_BANNER_ACTIVE_CLASSNAME)).toBeNull();
+          done();
+        });
+
+        it('after clicking yes or no and receiving a new answer, it should still be possible to give the same answer', async done => {
```

Comment:
Without resetting `isUseful`, we can't press "no" or "yes" on both the first banner and the next banner.